### ISSUE TITLE
rename-teams not restricted to main team

### ISSUE THREAD

**Issue by eruvanos** [concourse/concourse#2492]
# Bug Report

According to the [docs](https://concourse-ci.org/managing-teams.html#fly-rename-team) only the main team should be allowed to rename teams.

```
fly -t main set-team -n abc --basic-auth-username=abc --basic-auth-password=abc
fly -t abc login -c <concourse-url> -n abc

fly -t abc rename-team -o abc -n cde
# THIS SHOULD FAIL, BUT IT DOES NOT
```

The following can also be handy:

* Concourse version: 3.14.1
* Deployment type (BOSH/Docker/binary): BOSH
* Infrastructure/IaaS: AWS
* Browser (if applicable): commandline
* Did this used to work? -

**Comment by eruvanos** [comment]
It looks like the api works like intended. We found [tests](https://github.com/concourse/atc/blob/2119d034f6498242236369931eeef50c1fc91697/api/teams_test.go#L415) for the [current behaviour](https://github.com/concourse/atc/blob/2119d034f6498242236369931eeef50c1fc91697/api/teamserver/rename.go#L18).

We have a service broker that allows a concourse teams to be created and it is not possible for the broker to know if the team name (that is the Unique Primary identification) has been changed. IMHO only an admin must be allowed to do this as described in the documentation, unless `GUID` uniquely identifies a team and not a `team name`.
